curwmysqladapter/mysqladapter.py

```diff
@@ -100,18 +104,24 @@ def get_event_id(self, meta_data):
 
         m.update(json.dumps(hash_data, sort_keys=True).encode("ascii"))
         possible_id = m.hexdigest()
+        connection = self.pool.get()
         try:
-            with self.connection.cursor() as cursor:
+            with connection.cursor() as cursor:
                 sql = "SELECT 1 FROM `run` WHERE `id`=%s"
 
                 cursor.execute(sql, possible_id)
                 is_exist = cursor.fetchone()
                 if is_exist is not None:
                     event_id = possible_id
-        except Exception as e:
-            traceback.print_exc()
-        finally:
             return event_id
+        except Exception as ex:
+            error_message = 'Error in retrieving event_id for meta data: %s.' % meta_data
+            #TODO logging and raising is considered as a cliche' and bad practice.
```

Comment:
👍 
@madumalt should we consider both of them as bad practice. At least we should raise it right!. Otherwise, how will end user handle any failures as adapter level ?